### PR TITLE
Add pop operation

### LINKED ISSUE #125: [vm] Implement opcode Pop
Detail
Implement opcode Pop

Reference
https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go

### PR DESCRIPTION
**Pull request by hea9549**
resolved: #125 

details: implement pop opcode

I made a pop of vm. But I don't know if this is necessary.

- [x] Test case

### CODE REVIEW

File: vm/vm.go [hunk 1]
```diff
@@ -200,8 +201,11 @@ func (not) hex() []uint8 {
 	return []uint8{uint8(opcode.NOT)}
 }
 
-// TODO: implement me w/ test cases :-)
 func (pop) Do(stack *stack, _ asmReader) error {
+	if stack.len() == 0 {
+		return ErrStackEmpty
+	}
+	_ = stack.pop()
```
Comment (boohyunsik) [inline vm/vm.go:204]: if pop retuns err, why don't you return err?

Comment (hea9549) [reply vm/vm.go:204]: Do you mean stack.pop() throws an error? stack.pop() does not return an error.

Comment (boohyunsik) [reply vm/vm.go:204]: ah, I misunderstood :-D

**Review by boohyunsik — APPROVED** [review]
LGTM :-)

**Comment by hihiboss** [comment]
Is it really fine just removing item??

@hea9549 @boohyunsik  @zeroFruit @leesd556 @junbeomlee

**Comment by leesd556** [comment]
@hihiboss 
I think it should be discussed especially with the compiler!

1. Is it ok just pop without error handling?
 -> EVM don't handle that error. why?

2. Is it ok just removing item ?
 -> EVM uses `intPool`, a temporary stack storage. why?

I'll register it to issue!

**Review by AgwaB — CHANGES_REQUESTED** [review]
@hihiboss 
I think it should be discussed especially with the compiler!

1. Is it ok just pop without error handling?
 -> EVM don't handle that error. why?

2. Is it ok just removing item ?
 -> EVM uses `intPool`, a temporary stack storage. why?

I'll register it to issue!

**Comment by hihiboss** [comment]
@leesd556 

Ok. Thank you!

**Comment by hea9549** [comment]
> @hihiboss
> I think it should be discussed especially with the compiler!
> 
> 1. Is it ok just pop without error handling?
>    -> EVM don't handle that error. why?
> 2. Is it ok just removing item ?
>    -> EVM uses `intPool`, a temporary stack storage. why?
> 
> I'll register it to issue!

1. I submit issue about this #150 

2. I found lot's of article about intpool. In my opinion, an intpool is simply used for reuse. also written in evm's intpool.go
```
// intPool is a pool of big integers that
// can be reused for all big.Int operations.
```
It is used to reuse objects without creating new ones.

**Review by AgwaB — APPROVED** [review]

**Pull request merged**